### PR TITLE
chore(experiments HogQL): migrate set experiments UI

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Changes
^^

## How did you test this code?

**Metric display**
- [x] Trends main metric displayed correctly
- [x] Exposure main metric displayed correctly
- [x] Exposure metric displayed correctly

**Metric type switcher**
- [x] Sets default trend query
- [x] Sets default funnel query

**Funnel metric modal**
- [x] Series selector
- [x] Aggregating by
- [x] Conversion window limit value
- [x] Conversion window limit unit
- [x] Attribution type
- [x] Filter test accounts
- [x] Preview loads correctly
- [x] "Cancel" discards changes
- [x] "Save" persists changes

**Trend metric modal**
- [x] Series selector
- [x] Series math selector
- [x] Filter test accounts
- [x] Preview loads correctly
- [x] "Cancel" discards changes
- [x] "Save" persists changes

**Trend exposure metric modal**
- [x] Series selector
- [x] Series math selector
- [x] Filter test accounts
- [x] Preview loads correctly
- [x] "Cancel" discards changes
- [x] "Save" persists changes
- [x] "Reset" deletes the exposure query

**Secondary funnel metric modal**
- [x] Name input
- [x] Series selector
- [x] Aggregating by
- [x] Conversion window limit value
- [x] Conversion window limit unit
- [x] Attribution type
- [x] Filter test accounts
- [x] Preview loads correctly
- [x] "Cancel" discards changes
- [x] "Save" persists changes

**Secondary trend metric modal**
- [x] Name input
- [x] Series selector
- [x] Series math selector
- [x] Filter test accounts
- [x] Preview loads correctly
- [x] "Cancel" discards changes
- [x] "Save" persists changes

**Results**
- [x] Funnel results match up
- [x] Trend results match up
- [x] "Explore results" matches up

**Secondary metrics table**
- [x] The "+" button opens a new form
- [x] The edit button opens an edit form
- [x] The Delete button deletes the metric
- [x] The chart button displays the metric chart

### CODE REVIEW

**Review by danielbachhuber — APPROVED** [review]
Nice work!

I did a full read-through of the diff and didn't see any egregious errors. I left comments where I noticed things.

I also asked ChatGPT for its opinion (according to the [Maslow's pyramid of code review](https://www.dein.fr/posts/2015-02-18-maslows-pyramid-of-code-review)). Here's what it said: https://chatgpt.com/share/6737a836-5248-8011-9478-1eea8ebd187a

I checked out the code locally, without the feature flag enabled. I was able to create the experiment, assign a trend goal, and launch the experiment just fine. When I tried adding a secondary metric, I saw this error:

CleanShot 2024-11-15 at 12 11 33@2x

I was able to load an existing trend experiment just fine. I saw my results, the secondary metrics, variants, and cumulative exposures. I wasn't able to delete an existing secondary metric, however.

Looking forward to seeing this live!

File: ee/clickhouse/views/experiments.py [hunk 1]
```diff
@@ -301,7 +301,8 @@ def create(self, validated_data: dict, *args: Any, **kwargs: Any) -> Experiment:
 
         feature_flag_key = validated_data.pop("get_feature_flag_key")
 
-        properties = validated_data["filters"].get("properties", [])
+        # properties = validated_data["filters"].get("properties", [])
```
[the remark this answers is not in the document]
Comment (jurajmajerik) [reply ee/clickhouse/views/experiments.py:304]: Oh yeah, this is some legacy thing, I checked with Neil if I can remove it (I've cleaned up this comment now)

File: frontend/src/scenes/experiments/experimentLogic.tsx [hunk 2]
```diff
@@ -821,17 +839,20 @@ export const experimentLogic = kea<experimentLogicType>([
...
-                            const response: ExperimentResults = await api.create(
-                                `api/projects/${values.currentTeamId}/query`,
-                                { query }
-                            )
+                            const response = await performQuery(queryWithExperimentId, undefined, refresh)
 
                             return {
                                 ...response,
                                 fakeInsightId: Math.random().toString(36).substring(2, 15),
+                                // @ts-expect-error
```
[the remark this answers is not in the document]
Comment (jurajmajerik) [reply frontend/src/scenes/experiments/experimentLogic.tsx]: Cleaned up now.

**Comment by jurajmajerik** [comment]
Thanks for the thorough review @danielbachhuber!

I've fixed a few remaining issues and tested all the flows today. We're good to go, just waiting for the tests to pass 👀

**Pull request merged**